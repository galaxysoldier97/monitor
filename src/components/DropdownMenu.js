import React, { useState } from 'react';
import { IconButton, Menu } from '@material-ui/core';
import { MoreVert } from '@material-ui/icons';
import PropTypes from 'prop-types';

const DropdownMenu = ({children, hideEmpty = true}) => {
  const [anchorEl, setAnchorEl] = useState(null);
  const menuOpen = Boolean(anchorEl);

  const handleMenu = event => setAnchorEl(event.currentTarget);
  const handleClose = () => setAnchorEl(null);

  const menuItems = children(handleClose);
  const hideIfNullItems = hideEmpty && menuItems === null;
  if (hideIfNullItems || !menuItems || menuItems.length === 0) {
    return null;
  }

  return (
    <div>
      <IconButton color="inherit" aria-haspopup="true" onClick={handleMenu}>
        <MoreVert />
      </IconButton>
      <Menu anchorEl={anchorEl} open={menuOpen} onClose={handleClose}>
        {menuItems}
      </Menu>
    </div>
  );
};

DropdownMenu.propTypes = {
  children: PropTypes.func.isRequired,
  hideEmpty: PropTypes.bool,
};

export default DropdownMenu;
