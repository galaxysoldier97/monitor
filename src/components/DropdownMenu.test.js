import React from 'react';
import renderer from 'react-test-renderer';
import DropdownMenu from './DropdownMenu';

test('Dropdown menu snapshot', () => {
  const component = renderer.create(<DropdownMenu children={() => "test"} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
