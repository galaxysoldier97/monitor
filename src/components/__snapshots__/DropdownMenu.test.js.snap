// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown menu snapshot 1`] = `
<div>
  <button
    aria-haspopup="true"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </span>
  </button>
</div>
`;
